ecosystem/sep-0010.md

```diff
@@ -24,10 +24,16 @@ The authentication flow is as follows:
 1. The client verifies that the transaction has an invalid sequence number 0.  This is extremely important to ensure the transaction isn't malicious.
 1. The client signs the transaction using the secret key(s) of signers for the user's Stellar account
 1. The client submits the signed challenge back to the server using [`token`](#token) endpoint
-1. The server gets the signers of the user's account
-1. The server verifies the signatures
-1. The server verifies the weight provided by the signers meets the required threshold(s), if any
-1. If the signature checks out, the server responds with a [JWT](https://jwt.io) that represents the user's session
+1. The server checks that the user's account exists
+1. If the user's account exists:
+  1. The server gets the signers of the user's account
+  1. The server verifies the client signatures count is one or more;
+  1. The server verifies the client signatures on the transaction are signers user's account;
+  1. The server verifies the weight provided by the signers meets the required threshold(s), if any
+  1. If the signatures checks out, the server responds with a [JWT](https://jwt.io) that represents the user's session
+1. If the user's account does not exist (optional):
+  1. The server verifies the client signature count is one
+  1. The server verifies the client signature is correct for the master key of the account
```

Comment:
Maybe don't mention account here since it doesn't exist / maybe shouldn't exist? Not sure if that gets confusing though.

Reply:
I think it gets confusing, also because even for an account that doesn't exist we still refer to it as an account in the API. For example the URL is still `?account=<account-that-doesn't-exist>` 😬. So I think this line is just part of the problem, not the problem.